### PR TITLE
[flink] Remove scan.push-down support which should be supported by Flink

### PR DESCRIPTION
**Pull request by JingsongLi**
### Purpose

The scan.push-down makes source codes very complicate, it should be supported by Flink SQL instead of Paimon.

In this PR, we remove it, if the user have requirement, he should ask Flink community to support this.

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java [hunk 1]
```diff
@@ -18,68 +18,31 @@
...
 import static org.apache.paimon.utils.Preconditions.checkState;
 
 /**
- * Table source to create {@link StaticFileStoreSource} or {@link ContinuousFileStoreSource} under
- * batch mode or change-tracking is disabled. For streaming mode with change-tracking enabled and
- * FULL scan mode, it will create a {@link
- * org.apache.flink.connector.base.source.hybrid.HybridSource} of {@code
- * LogHybridSourceFactory.FlinkHybridFirstSource} and kafka log source created by {@link
- * LogSourceProvider}.
+ * A {@link BaseDataTableSource} implements {@link SupportsStatisticReport} and {@link
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java:41]: Change BaseDataTableSource to DataTableSource?

File: docs/layouts/shortcodes/generated/flink_connector_configuration.html [hunk 2]
```diff
@@ -116,12 +116,6 @@
             <td>Integer</td>
             <td>Define a custom parallelism for the scan source. By default, if this option is not defined, the planner will derive the parallelism for each statement individually by also considering the global configuration. If user enable the scan.infer-parallelism, the planner will derive the parallelism by inferred parallelism.</td>
         </tr>
-        <tr>
```
Comment (wwj6591812) [inline docs/layouts/shortcodes/generated/flink_connector_configuration.html:119]: Cut scan.push-down in ContinuousFileStoreITCase.java.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java (hunk 1 above)
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java:41]: It is a `BaseDataTableSource`, ref to self is meaningless.

File: paimon-flink/paimon-flink-1.15/src/main/java/org/apache/paimon/flink/source/DataTableSource.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DynamicTableFactory;
+
+import javax.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+
+/** A {@link BaseDataTableSource} for Flink 1.15. */
+public class DataTableSource extends BaseDataTableSource {
```
[the remark this answers is not in the document]
Comment (JingsongLi) [reply paimon-flink/paimon-flink-1.15/src/main/java/org/apache/paimon/flink/source/DataTableSource.java:36]: It is a `BaseDataTableSource`, ref to self is meaningless.

[the remark this answers is not in the document]
Comment (wwj6591812) [reply paimon-flink/paimon-flink-1.15/src/main/java/org/apache/paimon/flink/source/DataTableSource.java:36]: OK

**Comment by wwj6591812** [comment]
+1

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java (hunk 1 above)
Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java:41]: OK

File: paimon-flink/paimon-flink-1.15/src/main/java/org/apache/paimon/flink/source/DataTableSource.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (wwj6591812) [reply paimon-flink/paimon-flink-1.15/src/main/java/org/apache/paimon/flink/source/DataTableSource.java:36]: OK

**Pull request merged**